src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/cosmic/CosmicFuncotationFactory.java

```diff
@@ -0,0 +1,387 @@
+package org.broadinstitute.hellbender.tools.funcotator.dataSources.cosmic;
+
+import com.google.common.annotations.VisibleForTesting;
+import htsjdk.tribble.Feature;
+import htsjdk.variant.variantcontext.VariantContext;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.broadinstitute.hellbender.engine.ReferenceContext;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.tools.funcotator.DataSourceFuncotationFactory;
+import org.broadinstitute.hellbender.tools.funcotator.Funcotation;
+import org.broadinstitute.hellbender.tools.funcotator.dataSources.TableFuncotation;
+import org.broadinstitute.hellbender.tools.funcotator.dataSources.gencode.GencodeFuncotation;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.nio.file.Path;
+import java.sql.*;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+/**
+ * Factory for creating {@link Funcotation}s by handling a SQLite database containing information from COSMIC
+ * (http://cancer.sanger.ac.uk/cosmic/download).
+ *
+ * This is a high-level object that interfaces with the internals of {@link org.broadinstitute.hellbender.tools.funcotator.Funcotator}.
+ * Created by jonn on 12/16/17.
+ */
+public class CosmicFuncotationFactory extends DataSourceFuncotationFactory {
+
+    //==================================================================================================================
+    // Public Static Members:
+
+    //==================================================================================================================
+    // Private Static Members:
+
+    /** Standard Logger.  */
+    protected static final Logger logger = LogManager.getLogger(CosmicFuncotationFactory.class);
+
+    /**
+     * Regular expression for matching a genome position in a Cosmic record.
+     */
+    @VisibleForTesting
+    static final Pattern GENOME_POSITION_REGEX =  Pattern.compile("(\\d+):(\\d+)-(\\d+)");
+
+    /**
+     * Regular expression for matching a protein position in a Cosmic record.
+     */
+    @VisibleForTesting
+    static final Pattern PROTEIN_POSITION_REGEX =  Pattern.compile("[pP]\\.\\(?[A-Z](\\d+)\\)?[A-Z]?_?(?:[A-Z]?(\\d+)?.*)?");
+
+    /**
+     * Placeholder contig name for protein positions.
+     */
+    private static final String PROTEIN_CONTIG = "P";
+
+    /**
+     * Expected number of columns in the Cosmic table in the database.
+     */
+    private static final int EXPECTED_NUM_FIELDS = 35;
```

Comment:
This is only used for pre-allocation.  It doesn't affect any core processing.

In fact, this is now vestigial for COSMIC, but I've left it in so that I can pull out a generic database data source.

Reply:
Removed and created a new branch to capture the database reader fundamentals.